app/views/questionnaire.py

```diff
@@ -208,13 +212,22 @@ def get_thank_you(eq_id, form_type):  # pylint: disable=unused-argument
 
     if session_data.submitted_time:
         schema = load_schema_from_params(session_data.eq_id, session_data.form_type)
-        metadata_context = build_metadata_context_for_survey_completed(vars(session_data))
+        metadata_context = build_metadata_context_for_survey_completed(session_data)
+
+        view_submission_url, view_submission_duration = None, 0
+        if session_data.is_viewable_submission_stored:
```

Comment:
I think we should drop `is_viewable_submission_stored` and rely on checking the submitted time against the view submission duration. If nothing has gone wrong then it would work in the same way. If it hasn't been stored then we can error or redirect back to the thank-you page.